### PR TITLE
check for actionable reviews with DuplicateEP errors

### PR DESCRIPTION
**Pull request by tomas-nava**
Connects #11081 

### Description
Checks for `HigherLevelReview` and `SupplementalClaim` objects with `DuplicateEP` errors that may need manual resolution, and reports any it finds to the `#appeals-job-alerts` slack channel.

### CODE REVIEW

File: spec/services/reviews_with_duplicate_ep_error_checker_spec.rb [hunk 1]
```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+
+require "support/vacols_database_cleaner"
```
Comment (pkarman) [inline spec/services/reviews_with_duplicate_ep_error_checker_spec.rb]: Does this test touch FACOLS? I don't see it. If not, we can just use the `:postgres` cleaner which is faster.

File: app/services/reviews_with_duplicate_ep_error_checker.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    problem_reviews(reviews: HigherLevelReview.where(ERROR_SELECTOR), modifier: "030")
+  end
+
+  def problem_supplemental_claims
+    problem_reviews(reviews: SupplementalClaim.where(ERROR_SELECTOR), modifier: "040")
+  end
+
+  def problem_reviews(reviews:, modifier:)
+    reviews.select do |review|
+      review.veteran.end_products.select do |ep|
```
Comment (tomas-nava) [inline app/services/reviews_with_duplicate_ep_error_checker.rb]: Reminder to self to add safe navigators to this method.

File: spec/services/reviews_with_duplicate_ep_error_checker_spec.rb [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+    describe "#call" do
+      it "does not build a report" do
+        subject.call
+        expect(subject.report?).to be_falsey
+      end
+    end
+  end
+
+  describe ".slack_channel" do
```
[the remark this answers is not in the document]
Comment (tomas-nava) [reply spec/services/reviews_with_duplicate_ep_error_checker_spec.rb]: That's historic baggage from the last time I created a checker job and forgot to make the slack channel method public. But now it's defined in the parent class so not really needed.

File: app/services/reviews_with_duplicate_ep_error_checker.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+  def problem_higher_level_reviews
+    problem_reviews(reviews: HigherLevelReview.where(ERROR_SELECTOR), modifier: "030")
+  end
+
+  def problem_supplemental_claims
+    problem_reviews(reviews: SupplementalClaim.where(ERROR_SELECTOR), modifier: "040")
+  end
+
+  def problem_reviews(reviews:, modifier:)
```
[the remark this answers is not in the document]
Comment (tomas-nava) [reply app/services/reviews_with_duplicate_ep_error_checker.rb]: fixed with [this commit](https://github.com/department-of-veterans-affairs/caseflow/pull/12221/commits/69a643ad6e7f3f61f9a88dc8fc335f03df8c4115)

File: spec/services/reviews_with_duplicate_ep_error_checker_spec.rb [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+          status_type_code: "CAN",
+          last_action_date: 3.days.ago.to_formatted_s(:short_date),
+          claim_type_code: "030BLAH",
+          error_text: error_text
+        },
+        {
+          file_number: "000_000_002",
+          review_type: :higher_level_review,
+          status_type_code: "CLR",
+          last_action_date: 2.days.ago.to_formatted_s(:short_date),
```
[the remark this answers is not in the document]
Comment (tomas-nava) [reply spec/services/reviews_with_duplicate_ep_error_checker_spec.rb]: fixed with [this commit](https://github.com/department-of-veterans-affairs/caseflow/pull/12221/commits/b0bc9f7cec4a78162af26848b491ebc4c991a07a)

File: spec/services/reviews_with_duplicate_ep_error_checker_spec.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (tomas-nava) [reply spec/services/reviews_with_duplicate_ep_error_checker_spec.rb]: removed with [this commit](https://github.com/department-of-veterans-affairs/caseflow/pull/12221/commits/d84ff7b8ad6f4d95ee03cfade6438331cfdf7ec1)

File: spec/services/reviews_with_duplicate_ep_error_checker_spec.rb (hunk 1 above)
Comment (tomas-nava) [reply spec/services/reviews_with_duplicate_ep_error_checker_spec.rb]: fixed with [this commit](https://github.com/department-of-veterans-affairs/caseflow/pull/12221/commits/77d787803942d24dabea1901e0c5088c1f95695a)

File: app/services/reviews_with_duplicate_ep_error_checker.rb (hunk 2 above)
Comment (tomas-nava) [reply app/services/reviews_with_duplicate_ep_error_checker.rb]: added in [this commit](https://github.com/department-of-veterans-affairs/caseflow/pull/12221/commits/dcb3b0ca74366656a35c2fc768ff76e3263ad83f)

Comment (tomas-nava) [reply app/services/reviews_with_duplicate_ep_error_checker.rb]: and reverted in [this commit](https://github.com/department-of-veterans-affairs/caseflow/pull/12221/commits/f91a878eed5d6724ed736a8afe1bb897cd284bed); not really necessary and it was attracting unwanted attention from codeclimate.

**Pull request merged**